src/main/daml/Daml/Finance/Instrument/Bond/Util.daml

```diff
@@ -90,6 +89,22 @@ createCallableBondClaims dateToTime schedule periodicSchedule useAdjustedDatesFo
                 Some Calendar -> addDays baseDate dayShift
               resetDate = adjustDate fixingCalendars fr.fixingDates.businessDayConvention
                 resetDateUnadj
+              regularRate = Observe fr.referenceRateId + Const couponRate
+              dcfAndNotional = scale dcf notionalAtPeriodEnd
+              regularRateClaim = scale regularRate dcfAndNotional
+              rateAfterCap = case capRate of
+                None -> regularRateClaim
+                Some cap -> cond capCondition capRateClaim regularRateClaim
+                  where
+                    capCondition = Lte (Const cap, regularRate)
```

Comment:
Fyi, you can write this as `Const cap <= regularRate`

Reply:
I fixed this as part of this PR: https://github.com/digital-asset/daml-finance/pull/779